src/base58.ts

```diff
@@ -30,7 +30,8 @@ export function decodeBase58Check(versionByteName, encoded) {
     );
   }
 
-  const expectedChecksum = calculateChecksum(payload);
+  // TODO: Investigate. This looks like a upstream typing bug, because empirically it works anyways.
```

Comment:
Can you use TS-TODO so it's easier for us to identify the todos related with this migration?